src/main/java/com/example/cats/controller/ProductController.java

```diff
@@ -0,0 +1,62 @@
+package com.example.cats.controller;
+
+
+import com.example.cats.dto.ProductDTO;
+import com.example.cats.service.ProductService;
+import jakarta.validation.Valid;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.NoSuchElementException;
+
+@RestController
+@RequestMapping("/api/products")
+public class ProductController {
+
+    private final ProductService service;
+
+    public ProductController(ProductService service) {
+        this.service = service;
+    }
+
+    // Create
+    @PostMapping
+    public ResponseEntity<ProductDTO> create(@Valid @RequestBody ProductDTO dto) {
+        ProductDTO created = service.create(dto);
+        return ResponseEntity.status(HttpStatus.CREATED).body(created);
+    }
+
+    // Read all
+    @GetMapping
+    public List<ProductDTO> all() {
+        return service.findAll();
+    }
+
+    // Read one
+    @GetMapping("/{id}")
+    public ResponseEntity<ProductDTO> getOne(@PathVariable Long id) {
+        return service.findById(id)
+                .map(ResponseEntity::ok)
+                .orElseGet(() -> ResponseEntity.notFound().build());
```

Comment:
tha't actually should be handled on the level of the service, you should throw the error on the level of service, in method findById, and your controller advice with error handling will catch this exeption please revisit the lecture -1